### ISSUE TITLE
Copies with a "-n " prefix

### ISSUE THREAD

**Issue by Guchman** [Morantron/tmux-fingers#47]
Hi! My setup is:
- osX
- iterm2/alacritty
- yank
- tmux 2.6
- gawk 4.2.1

Every time then I'd select something with fingers it goes with "-n  " prefix
f.e.
 
` 2> /dev/null)`
matchs as /dev/null[p]
after I press 'p' and paste selection to any editor it's looks like
`-n /dev/null
`
with weird -n prefix and new line at the end.
it's very annoying.
I've tried to enable debug.sh but it's doesn't do anything:(
I've tried multiple terminals. Also bash/zsh shells. Nothing changed.
When I've select with plain copy-mode - everything works ok.

**Comment by Morantron** [comment]
Can you check which tmux-fingers version do you have? You can check so by entering fingers-mode and then pressing <kbd>?</kbd>. In theory :tm: latest version ( 0.7.1 ) works on OSX, but who knows in practice :trollface:

**Comment by Guchman** [comment]
@Morantron 0.7.1

**Comment by Morantron** [comment]
Could you share your `.tmux.conf`? :pray: there were some changes recently in `@fingers-copy-command`, maybe the problem lies there

**Comment by Morantron** [comment]
Also running `/path/to/tmux-fingers/scripts/health-check.sh` can help troubleshooting sometimes

**Comment by Guchman** [comment]
@Morantron 
health-check output:
> /Users/nu.guchakov/.tmux/plugins/tmux-fingers/scripts/utils.sh: line 9: warning: setlocale: LC_CTYPE: cannot change locale (en_RU.UTF-8): Bad file descriptor

My .tmux.conf without irrelevant such as styles and shortcuts overrides:
`
set -g mouse on
set-option -g default-shell /bin/zsh
set -g prefix C-a
unbind C-b

set -g default-terminal "screen-256color"
set -g remain-on-exit on

setw    -g  mode-keys    vi
bind-key -T edit-mode-vi Up                send-keys -X history-up
bind-key -T edit-mode-vi Down              send-keys -X history-down
bind-key -T copy-mode-vi v                 send      -X begin-selection
bind-key -T copy-mode-vi [                 send-keys -X begin-selection
bind-key -T copy-mode-vi y                 send-keys -X copy-pipe-and-cancel "pbcopy"
bind-key -T copy-mode-vi MouseDragEnd1Pane send-keys -X copy-pipe-and-cancel "pbcopy"
bind-key -T copy-mode-vi Enter       
bind-key -T copy-mode-vi ]                 send-keys -X copy-selection\; display "Copied"

# List of plugins
set -g @plugin 'tmux-plugins/tpm'
set -g @plugin 'tmux-plugins/tmux-sensible'
set -g @plugin 'tmux-plugins/tmux-yank'
set -g @plugin 'tmux-plugins/tmux-resurrect'
set -g @plugin 'tmux-plugins/tmux-continuum'
set -g @continuum-boot 'on'
set -g @continuum-restore 'on'
set -g @continuum-boot-options 'fullscreen'

set -g @plugin 'tmux-plugins/tmux-open'
set -g @open 'C-o'
set -g @open-editor 'o'

#set -g @plugin "soyuka/tmux-current-pane-hostname"
set -g @plugin 'tmux-plugins/tmux-prefix-highlight'
set -g status-left  '#{prefix_highlight}'
# set -g status-right '[#h]:[%a %Y-%m-%d %H:%M]'
set -g @prefix_highlight_output_prefix '<'
set -g @prefix_highlight_output_suffix '>'
set -g @prefix_highlight_fg colour234
set -g @prefix_highlight_bg white
set -g @prefix_highlight_show_copy_mode 'on'
set -g @prefix_highlight_copy_mode_attr 'fg=black,bg=yellow,bold' # default is 'fg=default,bg=yellow'

set -g @plugin 'Morantron/tmux-fingers'
set -g @fingers-compact-hints 0
set -g @fingers-hint-format '#[fg=red,bold][%s]'

set -g @tpm_plugins "soyuka/tmux-current-pane-hostname"

# Initialize TMUX plugin manager (keep this line at the very bottom of tmux.conf)
run '~/.tmux/plugins/tpm/tpm'
`

**Comment by Morantron** [comment]
weiiirrrdd

you can use `debug.sh` to add traces to see what is happening

problem is probably in [this file](https://github.com/morantron/tmux-fingers/blob/master/scripts/fingers.sh)

You'd need to source it at the top of the file

```diff
#!/usr/bin/env bash

eval "$(tmux show-env -g -s | grep ^FINGERS)"

CURRENT_DIR="$( cd "$( dirname "${BASH_SOURCE[0]}" )" && pwd )"

source $CURRENT_DIR/hints.sh
source $CURRENT_DIR/utils.sh
source $CURRENT_DIR/help.sh
+source $CURRENT_DIR/debug.sh
```

And then add traces like this

```diff
function copy_result() {
  local result="$1"
  local hint="$2"

  tmux set-buffer "$result"

  if [[ "$OSTYPE" == "linux-gnu" ]]; then
    exec_prefix="nohup"
  else
    exec_prefix=""
  fi

  is_uppercase=$(echo "$input" | grep -E '^[a-z]+$' &> /dev/null; echo $?)

+ log "result: '$result'"

  if [[ $is_uppercase == "1" ]] && [ ! -z "$FINGERS_COPY_COMMAND_UPPERCASE" ]; then +  log "echo -n \"$result\" | IS_UPPERCASE=$is_uppercase HINT=$hint $exec_prefix       
    tmux run-shell -b "echo -n \"$result\" | IS_UPPERCASE=$is_uppercase HINT=$hint $exec_prefix $FINGERS_COPY_COMMAND_UPPERCASE > /dev/null"
  elif [ ! -z "$FINGERS_COPY_COMMAND" ]; then
    tmux run-shell -b "echo -n \"$result\" | IS_UPPERCASE=$is_uppercase HINT=$hint $exec_prefix $FINGERS_COPY_COMMAND > /dev/null"
  fi

  if [[ $HAS_TMUX_YANK = 1 ]]; then
    tmux run-shell -b "echo -n \"$result\" | $exec_prefix $tmux_yank_copy_command > /dev/null"
  fi
}
```

The results should show in `/path/to/tmux-fingers/fingers.log`

**Comment by Morantron** [comment]
Would it be possible that you have some alias set on `echo`?

What's the output of `$ alias | grep '^alias echo'` ?

**Comment by Guchman** [comment]
@Morantron empty:)

**Comment by Guchman** [comment]
It's weird.
echo -n "result" in fingers.sh works as echo "-n result"
in the terminal echo -n  works ok

**Comment by Guchman** [comment]
But 
```
 tmux run-shell -b "echo \"$result\\c\" | $exec_prefix $tmux_yank_copy_command > /dev/null"
```
works fine. I don't know the reason, but that seems to be workable solution on other systems to..

**Comment by Guchman** [comment]
Also where is weird spaces on copied line start.
maybe it'll be a good idea to trim them in copy function?

**Comment by Morantron** [comment]
Thanks for the research, fixed into 0.7.2 version :dancer:
